default.nix

```diff
@@ -299,15 +299,19 @@ let
       };
       fromImageFlag = l.optionalString (fromImage != "") "--from-image ${fromImage}";
       layerPaths = l.concatMapStringsSep " " (l: l + "/layers.json") ([customizationLayer] ++ layers);
-      image = pkgs.runCommand "image-${baseNameOf name}.json"
-      {
+      image = let
         imageName = l.toLower name;
+        imageTag =
+          if tag != null
+          then tag
+          else
+          l.head (l.strings.splitString "-" (baseNameOf image.outPath));
+      in pkgs.runCommand "image-${baseNameOf name}.json"
+      {
+        inherit imageName;
         passthru = {
-          imageTag =
-            if tag != null
-            then tag
-            else
-            l.head (l.strings.splitString "-" (baseNameOf image.outPath));
+          inherit imageTag;
+          imageRef = builtins.unsafeDiscardStringContext "${imageName}:${imageTag}"; 
```

Comment:
Could you add a comment to explain why we remove the string context?

(this builtin is not really famous and its doc hard to find)

Reply:
It's not "unsafe" in the sense that it is always unsafe to use, but more like a Rust `unsafe` block, where you shouldn't use carelessly without due attention and review. A string context is used to ensure that any store paths needed to build a string are carried along with it so if other derivations include them as inputs to a build, those dependancies will be properly captured.

For example, in the `copyTo` scripts, the context on the tag and name ensure that the OCI image is actually built and exists in the store for the script to make use of.

In this case, we are not building a derivation and we don't want to reference anything in the nix store, we just want a string that describes the image in a way other tools can understand, and ideally, we want it without having to build anything, and with minimal evaluation.

With that said, naming it `imageRefUnsafe` seems like a good idea just to ensure users understand that there is no guarantee that the underlying image will exist in the store. There are several instances in my workflow where this reference is useful though. For example. building a json job description for a cluster scheduler. In order for the job to execute successfully, the image has to exist in our image registry, and whether or not it exists in the nix store of the user is irrelevant.

In another instance, I check in CI if OCI images already exist in our registry before executing a `copyToRegistry` just to avoid wasiting excessive build cycles in CI, and longer wait times for all the images we upload to finish.

Reply:
Forgive me, I couldn't hold back this shameless plug: https://divnix.github.io/std/explain/why-nix.html#nix-superpowers : 😉